builder/replica_pipeline.cc

```diff
@@ -146,8 +146,25 @@ class pipeline_replicator : public expr_visitor {
     buffers_emitted_.insert(bep->sym());
 
     std::string elem_size = print_expr_inlined(bep->elem_size());
-    (void)print_assignment_explicit(
-        name, "buffer_expr::make(ctx, \"", name, "\", /*rank=*/", bep->rank(), ", /*elem_size=*/", elem_size, ")");
+    if (bep->constant()) {
+      std::string const_name = name + "_const";
+      (void)print_assignment_explicit(
+          const_name, "std::make_shared<buffer<void, ", bep->rank(), ">>(/*rank=*/", bep->rank(), ", /*elem_size=*/", elem_size, ")");
+      for (std::size_t d = 0; d < bep->rank(); d++) {
+          std::string mn = print_expr_inlined(bep->dim(d).bounds.min);
+          std::string mx = print_expr_inlined(bep->dim(d).bounds.max);
```

Comment:
This should save all the dim metadata, including stride and fold factor. It's possible and seems useful to encode information this way, e.g. broadcasting with stride 0, or "modular broadcasting" with a fold factor.

Also, I'm wondering if it should use the buffer object, not buffer_expr: https://github.com/dsharlet/slinky/blob/b6b6cd9eb5ac437ea1d1838e03b85dde192e9241/builder/pipeline.h#L85. I can see arguments for both:
- It makes more sense to use the buffer because if these exprs aren't constants, this won't compile.
- It makes more sense to use buffer_expr because that is more faithful to the original pipeline.
- Either way, these *should* be equal unless something has gone wrong somewhere.

Reply:
PTAL